lib/cliapp/src/Obelisk/CliApp/Process.hs

```diff
@@ -212,18 +212,28 @@ callCommand cmd = do
   liftIO $ Process.callCommand cmd
 
 withProcess
-  :: (MonadIO m, CliLog m, CliThrow e m, AsProcessFailure e, MonadFail m)
+  :: (MonadIO m, CliLog m, CliThrow e m, AsProcessFailure e, MonadFail m, MonadMask m)
   => ProcessSpec -> (Handle -> Handle -> m ()) -> m (Handle, Handle)
-withProcess process f = do -- TODO: Use bracket.
-  -- FIXME: Using `withCreateProcess` here leads to something operating illegally on closed handles.
-  (_, Just out, Just err, p) <- createProcess $ overCreateProcess
-    (\x -> x { std_out = CreatePipe , std_err = CreatePipe }) process
-
-  f out err  -- Pass the handles to the passed function
-
-  waitForProcess p >>= \case
-    ExitSuccess -> return (out, err)
-    ExitFailure code -> throwError $ review asProcessFailure $ ProcessFailure (Process.cmdspec $ _processSpec_createProcess process) code
+withProcess process f =
+  bracketOnError
+    (createProcess $ overCreateProcess
+      (\x -> x { std_out = CreatePipe , std_err = CreatePipe }) process
+    )
+    (liftIO . Process.cleanupProcess)
+    (\(_, Just out, Just err, p) -> do
```

Comment:
If we don't get two `Just ..` values then this will fail with an ugly pattern failure exception, we should handle that even if it's "impossible".

Reply:
Fixed